fix(AWS CloudFront): Recognize `behavior.TrustedKeyGroups` in schema

preshetin opened the pull request:
Closes: #9595

This PR is built on top of https://github.com/serverless/serverless/pull/9598 addressing https://github.com/serverless/serverless/pull/9598#issuecomment-874777145 also:

> In addition - is it possible to reference these groups e.g. with CF intrinsic functions such as Ref, etc? If yes, we should also allow schema to accept them.

I found [one example](https://github.com/awslabs/amplify-video/issues/249#issuecomment-831432822) were `Ref` is used when defining a trusted key group. But I doubt it _has_ to be a ref since the example resource in the initial issue (see it's `serverless.yrml`) uses a value like `abcdef ` which is obviously not using `Ref`. So I think we should not add intrinsic function constraint to that piece of schema.

GrahamCampbell commented:
Can we get some test coverage here?

preshetin commented:
> About support for `!Ref` - based on your example, it looks like it should be possible to provide the `TrustedKeyGroups` with `Ref` CF function, correct? If that's the case, I think we should allow that notation in our schema - in current form it will produce a warning/error when CF function is provided.

Hey @pgrzesik, looks like we have to provide support for Ref CF intristic functions. I'm not 100% sure though.  I've added support to this, so I can confirm the validation shows green light for config like this:

```
functions:
  hello:
    handler: handler.hello
    events:
      - cloudFront:
          behavior:
            TrustedKeyGroups:
              - foo
              - !Ref bar
```



> Can we get some test coverage here?

I believe we don't need test support for config schema declarations. This job gets delegated to AJV, so there is no need to test things that are declared

pgrzesik commented on lib/plugins/aws/package/compile/events/cloudFront.js:
```diff
@@ -163,6 +163,16 @@ class AwsCompileCloudFrontEvents {
           enum: ['allow-all', 'redirect-to-https', 'https-only'],
         },
+        TrustedKeyGroups: {
+          anyOf: [
```
Is that `anyOf` needed here? I think it's an unnecessary wrapper.

preshetin replied:
Nice catch @pgrzesik, I forgot to remove it

pgrzesik reviewed (changes requested):
Thanks @preshetin - looks great, I only have one minor suggestion and we should be ready to merge :+1:

preshetin commented:
@pgrzesik, I've addressed the comment that appeared after the first review. 

I tested the new changes and can confirm it works too, and now it's cleaner

pgrzesik reviewed (approved):
Thanks @preshetin, well done :clap:
